pkg/notary/registry.go

```diff
@@ -25,8 +25,11 @@ func parseReferenceCrane(ctx context.Context, ref string, registryClient images.
 		return nil, err
 	}
 
+	if err := registryClient.RefreshKeychainPullSecrets(ctx); err != nil {
+		return nil, errors.Wrapf(err, "failed to refresh image pull secrets")
+	}
```

Comment:
this cannot be exposed in the parse reference function, it is an internal operation.